hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -294,7 +294,7 @@ public R processExpiredEntry(Data key, R record, long expiryTime, long now) {
         records.remove(key);
         if (isEventsEnabled) {
             publishEvent(CacheEventType.EXPIRED, key, null, toEventData(record), false,
-                    IGNORE_COMPLETION, CacheRecord.EXPIRATION_TIME_NOT_AVAILABLE);
+                    IGNORE_COMPLETION, CacheRecord.EXPIRATION_TIME_NOT_AVAILABLE, null);
         }
```

Comment:
instead of passing null to the method, you can keep old method signature and inside it you can delegate all existing parameters + null for origin field. That seems more clear to me.